src/services/GraphsDFG.jl

```diff
@@ -342,6 +361,83 @@ Checks if the graph is not fully connected, returns true if it is not contiguous
 """
 hasOrphans(dfg::GraphsDFG)::Bool = !isFullyConnected(dfg)
 
+
+"""
+    $(SIGNATURES)
+Test if all elements of the string is a number:  Ex, "123" is true, "1_2" is false.
+"""
+allnums(str::S) where {S <: AbstractString} = occursin(Regex(string(["[0-9]" for j in 1:length(str)]...)), str)
+# occursin(r"_+|,+|-+", node_idx)
+
+isnestednum(str::S; delim='_') where {S <: AbstractString} = occursin(Regex("[0-9]+$(delim)[0-9]+"), str)
+
+function sortnestedperm(strs::Vector{<:AbstractString}; delim='_')
+  str12 = split.(strs, delim)
+  sp1 = sortperm(parse.(Int,getindex.(str12,2)))
+  sp2 = sortperm(parse.(Int,getindex.(str12,1)[sp1]))
+  return sp1[sp2]
+end
+
+function getFirstNumericalOffset(st::AS) where AS <: AbstractString
+  i = 1
+  while !allnums(st[i:i])  i+=1; end
+  return i
+end
+
+"""
+    $SIGNATURES
+
+Sort a variable list which may have nested structure such as `:x1_2` -- does not sort for alphabetic characters.
+"""
+function sortVarNested(vars::Vector{Symbol})::Vector{Symbol}
```

Comment:
Is this used in GraphsDFG? If not, do you mind it to a common utils file?

Reply:
think that should be better?